warehouse/bcm/backend_config.go

```diff
@@ -331,3 +334,20 @@ func deepCopy(src, dest interface{}) error {
 	}
 	return json.Unmarshal(buf, dest)
 }
+
+// TODO: use statsFactory
+func (s *BackendConfigManager) persistSSLFileErrorStat(
+	workspaceID, destType, destName,
+	destID, sourceName, sourceID,
+	errTag string,
+) {
+	tags := stats.Tags{
+		"workspaceId":   workspaceID,
+		"module":        "warehouse",
+		"destType":      destType,
+		"warehouseID":   misc.GetTagName(destID, sourceName, destName, misc.TailTruncateStr(sourceID, 6)),
+		"destinationID": destID,
+		"errTag":        errTag,
+	}
+	stats.Default.NewTaggedStat("persist_ssl_file_failure", stats.CountType, tags).Count(1)
```

Comment:
Can't we inject `stats` instead? We would also justify the method receiver here.

Reply:
I was thinking of injecting it in a separate PR.

Reply:
Added the changes.